.github/workflows/build-ants-windows.yml

```diff
@@ -0,0 +1,127 @@
+name: Build ANTs
+
+on:
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      git_ref:
+        description: 'Upstream ANTsX/ANTs commit ref to check out when building'
+        required: true
+        default: 'b37e8b56e4e518b0f2947f80d0e3701a5ddfe120'
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        os: [ windows-2019 ]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+    # first, rewrite OS names to be safe for github artifacts.
+    - name: names
+      run: |
+        OS="${{ matrix.os }}"
+        OS=$(echo "$OS" | sed s/://)
+        echo "ARTIFACT=$OS" >> $GITHUB_ENV
+    - uses: actions/checkout@v3
+      with:
+        repository: ANTsX/ANTs
+        # NB: This commit mirrors the state of the kousu/ANTs repository
+        #     at the time of https://github.com/kousu/ANTs/pull/5.
+        ref: ${{ github.event.inputs.git_ref || 'b37e8b56e4e518b0f2947f80d0e3701a5ddfe120' }}
+    # The step below sets an option inside the cmake file to use the `https` protocol, which addresses:
+    # https://github.blog/2021-09-01-improving-git-protocol-security-github/#no-more-unauthenticated-git
+    - name: Patch in fix for insecure git:// protocol
+      run: |
+        perl -pi -e 's/http instead." ON\)/http instead." OFF\)/g' SuperBuild.cmake
+    - name: cmake generate
+      run: |
+        mkdir antsbin
+        cd antsbin
+        # TODO: check if cmake picks up CMAKE_CXX_FLAGS without being told about it
+        # NB: `BUILD_TESTING=OFF` fixes https://github.com/ANTsX/ANTs/issues/1236
+        # NB: `-g "Unix Makefiles"` is needed to generate files for make (https://stackoverflow.com/a/59551107)
+        # NB: `CMAKE_CONFIGURATION_TYPES=Release` is needed on Windows to build the Release version (https://stackoverflow.com/a/20423820)
+        cmake \
+         -DBUILD_TESTING=OFF \
+         -DBUILD_SHARED_LIBS=OFF \
+         -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
+         -DZLIB_INCLUDE_DIR=D:\lib\zlib \
+         -DZLIB_LIBRARY=D:\lib\zlib\libz.dll.a \
+         -DCMAKE_BUILD_TYPE=Release \
+         -DCMAKE_CONFIGURATION_TYPES=Release \
+         ../
+        ls
+    - name: Add msbuild to PATH
+      uses: microsoft/setup-msbuild@v1.1
+      with:
+        msbuild-architecture: x64
+    - name: make
+      working-directory: antsbin
+      shell: cmd
+      run: |
+        dir
+        msbuild ALL_BUILD.vcxproj -property:Configuration=Release
```

Comment:
So, here's the current situation:

* When I [successfully built ANTs using Windows](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3682), I used `msbuild` from MS Visual Studio 2015 on my local machine. 
* However, the `windows-2019` GitHub Actions runner [only has MS Visual Studio 2019](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#visual-studio-enterprise-2019).
* Between 2015 and 2019, an upstream bug in ITK was [discovered](https://github.com/InsightSoftwareConsortium/ITK/issues/1967) and [fixed](https://github.com/InsightSoftwareConsortium/ITK/pull/2388).
* However, to get the fix, we have to bump the version of ITK built alongside ANTs to >=5.2.0 (see [this issue description](https://github.com/SuperElastix/SimpleElastix/pull/425) for someone in a similar predicament). 
* And, since we're using the ANTs SuperBuild, which `git clones` a stable release of ANTs, this probably means bumping the version of ANTs to one where ITK v5.2.0 is fetched.
* However, this has the unintended consequence of either causing a mismatch with the version of ANTs compiled for Windows and Linux/macOS, or forcing us to bump the version of ANTs used for Linux/macOS to match the one used for Windows.

However, [this comment](https://github.com/SuperElastix/SimpleElastix/pull/425#issuecomment-878516818) suggests a workaround:

> As a quick-fix you may simply remove both static lrintf function definitions from you copy of ITK/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h The function isn't being called by anyway, neither by ITK nor by elastix.

It's a bit cludgy, but I think I want to try applying the workaround to see if it allows us to build correctly.

Reply:
Using the updated ANTs (with the updated ITK) in [`07a466d` (#3)](https://github.com/spinalcordtoolbox/build_ANTs/pull/3/commits/07a466dc083c61199df19c67ae32d59e9ac08508) allows Windows to build fully using the `msbuild` from MS Visual Studio 2019.

I'm going to now try building from this ANTs commit across all of our platforms to see if everything goes OK. ([04a4a58](https://github.com/spinalcordtoolbox/build_ANTs/pull/3/commits/04a4a586a7c02fb8f3711c78a31203a9dca71cea))

Reply:
Is this summary right?

- The old ANTs doesn't build with the new VS 2019
- But the new ANTs trips over the numerical instability that we haven't ironed out (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3597, https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3194, https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2644)
- The old ANTs builds correctly with VS 2015 (and gives correct numerical results)?

What if you installed VS 2015 on the CI machine, then? There's a [`choco` package](https://community.chocolatey.org/packages/VisualStudioCommunity2015), and the CI machines come with `choco` installed.

Otherwise, the [numerical discrepencies](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/6454776240?check_suite_focus=true#step:5:228-) might be small enough to just ignore and maybe upgrading is the best? I vote for this because I detest maintaining out of date software (and if we used a more recent version then maybe it wouldn't be so hard to get away from vendoring ITK/ANTs/VTK at all). We would probably think about doing a major release if we did this.

Reply:
> Otherwise, the [numerical discrepencies](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/6454776240?check_suite_focus=true#step:5:228-) might be small enough to just ignore and maybe upgrading is the best? I vote for this because I detest maintaining out of date software (and if we used a more recent version then maybe it wouldn't be so hard to get away from vendoring ITK/ANTs/VTK at all). We would probably think about doing a major release if we did this.

I feel you here, and I largely agree? 

But, for context: In our last SCT meeting, Julien emphasized that (paraphrasing) "we have a working set of binaries right now, so making changes to them isn't really priority". So, I already feel a little guilty for spending time and energy finishing up a project that was explicitly said to _not_ be a priority. 

The only reason I started working on these build scripts in the first place was to clean up our old, unfinished draft PRs so that the work wouldn't get left behind. My hope was to quietly get the CI-building infrastructure up and running, and document it thoroughly on SCT's dev wiki. That way, if an issue were to arise in the future where an SCT dev needed to update ANTs (e.g. https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/2642), they could work with the existing build infrastructure.

Because of Julien's stance on this, I'd prefer to be able to set this up quietly and without much fuss? (i.e. not really making any changes that require big decision-making.)

> What if you installed VS 2015 on the CI machine, then? There's a [`choco` package](https://community.chocolatey.org/packages/VisualStudioCommunity2015), and the CI machines come with `choco` installed.

This seems like an interesting route to take!! Good thinking. Thank you for the idea. 😄

Reply:
Crisis averted, I think?

I was able to build ANTs on the `windows-2019` GitHub runner (while not bumping ANTs!) by patching in the workaround for the OpenJPEG library (see: https://github.com/spinalcordtoolbox/build_ANTs/runs/6456837755?check_suite_focus=true)

I've restarted the build, this time with all 5 OSs, and I _think_ we should end up with ANTs binaries built by CI that work for all of our platforms.